server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -623,7 +665,35 @@ public int compare(File o1, File o2)
           if (Ints.tryParse(segmentDir.getName()) == null) {
             continue;
           }
-
+          QueryableIndex queryableIndex = null;
+          try {
+            queryableIndex = IndexIO.loadIndex(segmentDir);
+          }
+          catch (IOException e) {
+            log.error(e, "Problem loading segmentDir from disk.");
+            isCorrupted = true;
+          }
+          if (isCorrupted) {
+            try {
+              File corruptSegmentDir = computeCorruptedFileDumpDir(segmentDir, schema);
+              log.info("Renaming %s to %s", segmentDir.getAbsolutePath(), corruptSegmentDir.getAbsolutePath());
+              FileUtils.copyDirectory(segmentDir, corruptSegmentDir);
+              FileUtils.deleteDirectory(segmentDir);
+            }
+            catch (Exception e1) {
+              log.error(e1, "Failed to rename %s", segmentDir.getAbsolutePath());
+            }
+            continue;
+          }
+          BasicFileAttributes attr = Files.readAttributes(segmentDir.toPath(), BasicFileAttributes.class);
+          if (attr.creationTime().toMillis() > latestCommitTime) {
```

Comment:
done. changed code to store a timestamp alongside commit metadata and use that instead of file system creation time.